README.md

```diff
@@ -23,6 +23,87 @@ gem install optimizely-sdk
 To access the Feature Management configuration in the Optimizely dashboard, please contact your Optimizely account executive.
 
 ### Using the SDK
+
+To use the SDK, the Optimizely instance can be initialized in three different ways as per your requirement. Ruby SDK also includes a factory class, OptimizelyFactory, which provides methods to create Optimizely instance with default configuration.
+
+ 1. Initialize Optimizely with a datafile. This datafile will be used as ProjectConfig throughout the life of Optimizely instance.
+    ~~~~~~~~~~~~
+    optimizely_instance = Optimizely::Project.new(datafile)
+2. Initialize Optimizely by providing an 'sdk_key'. This will initialize an HTTPConfigManager that makes an HTTP GET request to the URL ( formed using your provided sdk key and the default datafile CDN url template) to asynchronously download the project datafile at regular intervals and update ProjectConfig when a new datafile is recieved. 
+      ~~~~~~~~~~~~
+      optimizely_instance = Optimizely::OptimizelyFactory.create_default_instance_with_sdk_key('put_your_sdk_key_here')
+
+  The OptimizelyFactory class also exposes method to initialize Optimizely instance with sdk_key along with a default datafile. This hard-coded datafile will be used initially before any update.
+    
+    
+    optimizely_instance = Optimizely::OptimizelyFactory.create_default_instance_with_sdk_key_and_datafile('put_your_sdk_key_here', datafile)
+   
+ 3. Initialize Optimizely by providing a Config Manager that implements a 'get_config' method.You may use our HTTP Config Manager and customize it to your need. 
+    ~~~~~~~~~~~~
+    custom_config_manager = CustomConfigManager.new
+    optimizely_instance = Optimizely::OptimizelyFactory.create_default_instance_with_config_manager(custom_config_manager)
+
+
+#### HTTP Config Manager
+
+The HTTPConfigManager asynchronously polls for datafiles from a specified URL at regular intervals by making HTTP request.
+
+
+~~~~~~
+ http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
+        sdk_key: nil,
+        url: nil,
+        datafile: nil,
+        url_template: nil,
+        auto_update: nil,
+        polling_interval: nil,
+        start_by_default: nil,
+        blocking_timeout: nil,
+        logger: nil,
+        error_handler: nil,
+        skip_json_validation: false,
+        notification_center: notification_center
+      )
+~~~~~~   
+**Note**: One of the sdk_key or url must be provided. When both are provided, url takes the preference.
```

Comment:
Suggest revising note to:

**Note:** You must provide either the `sdk_key` or URL. If you provide both, the URL takes precedence.